pvlib/pvsystem.py

```diff
@@ -852,6 +852,67 @@ def i_from_v(self, resistance_shunt, resistance_series, nNsVth, voltage,
         return i_from_v(resistance_shunt, resistance_series, nNsVth, voltage,
                         saturation_current, photocurrent)
 
+    def get_ac(self, model, p_dc, v_dc=None):
+        r"""Calculates AC power from p_dc using the inverter model indicated
+        by model and self.inverter_parameters.
+
+        Parameters
+        ----------
+        model : str
+            Must be one of 'sandia', 'adr', or 'pvwatts'.
+        p_dc : numeric, or tuple, list or array of numeric
+            DC power on each MPPT input of the inverter. Use tuple, list or
+            array for inverters with multiple MPPT inputs. If type is array,
+            p_dc must be 2d with axis 0 being the MPPT inputs. [W]
+        v_dc : numeric, or tuple, list or array of numeric
+            DC voltage on each MPPT input of the inverter. Required when
+            model='sandia' or model='adr'. Use tuple, list or
+            array for inverters with multiple MPPT inputs. If type is array,
+            v_dc must be 2d with axis 0 being the MPPT inputs. [V]
+
+        Returns
+        -------
+        power_ac : numeric, or tuple of numeric
+            AC power output for the inverter. [W]
+
+        Raises
+        ------
+        ValueError
+            If model is not one of 'sandia', 'adr' or 'pvwatts'.
+
+        See also
+        --------
+        pvlib.inverter.sandia
+        pvlib.inverter.sandia_multi
+        pvlib.inverter.adr
+        pvlib.inverter.pvwatts
+        pvlib.inverter.pvwatts_multi
+        """
+        model = model.lower()
+        if model == 'sandia':
+            if self.num_arrays > 1:
+                p_dc = self._validate_per_array(p_dc)
```

Comment:
in other PVSystem methods we perform this validation independent of how many arrays exist, so I think it should move above the `if model` block.

The `v_dc` test should also move above the model block, but within a `if v_dc is not None` block.

Reply:
I can try that, but `_validate_per_array` always returns a tuple, which the inverter functions aren't set up to handle, e.g. `inverter.sandia` can't accept `p_dc` as a tuple.

I'm viewing `get_ac` as replacing the 5 inverter methods, not as a wrapper around those methods. Would you prefer that `get_ac` calls the `PVSystem` methods instead of the `inverter` functions?

Reply:
Ok, I misunderstood. Your method is good.